### ISSUE TITLE
Custom GeneratorStrategy  effets on Daos method name.

### ISSUE THREAD

**Issue by SightStudio** [jOOQ/jOOQ#15926]
### Your question

When Generating DSL from database
I create customGenerator that adds "J" prefix to table class to tell this class is generated by jOOQ (like QueryDSL)

```java
public class JPrefixGeneratorStrategy extends DefaultGeneratorStrategy {

    @Override
    public String getJavaClassName(final Definition definition, final Mode mode) {
        if (mode == Mode.DEFAULT || mode == Mode.POJO || mode == Mode.RECORD) {
            return "J" + super.getJavaClassName(definition, mode);
        }

        return super.getJavaClassName(definition, mode);
    }
}
```

but when i added prefix setting like above, 
generated dao class also has "J" prefix for each column.

```java
public class AddressDao extends DAOImpl<AddressRecord, Address, Long> {

    /**
     * I expected this would be "fetchRangeOfAddressId"
     */
    public List<Address> fetchRangeOfJAddressId(Long lowerInclusive, Long upperInclusive) {
        return fetchRange(JAddress.ADDRESS.ADDRESS_ID, lowerInclusive, upperInclusive);
    }

    public List<Address> fetchByJAddressId(Long... values) {
        return fetch(JAddress.ADDRESS.ADDRESS_ID, values);
    }

    public Address fetchOneByJAddressId(Long value) {
        return fetchOne(JAddress.ADDRESS.ADDRESS_ID, value);
    }

}
```

How can I get both a table class with J prefix and a dao without J prefixed method?

---- 
my little investigation... (this might be wrong. because I can only see the trees in the forest for now)

1. custom DefaultGeneratorStrategy can't tell this method is used for table class generation only or including dao column generation
maybe add API like ```protected String getJavaColumnName(...)``` can solve this issue

2. or like EmbeddableType...  use ```colMemberUC```for generation

https://github.com/jOOQ/jOOQ/blob/main/jOOQ-codegen/src/main/java/org/jooq/codegen/JavaGenerator.java#L5326

<img width="1763" alt="스크린샷 2023-12-09 오후 8 56 56" src="https://github.com/jOOQ/jOOQ/assets/50672087/a04db9f8-3fb1-43de-9bbf-0a21ae431d65">

----

sorry for bad English 😄 

### jOOQ Version

jOOQ oss 3.18.7

### Database product and version

not database specific

### Java Version

not jvm specific

### OS Version

not os specific

### JDBC driver name and version (include name if unofficial driver)

not JDBC driver specific

**Comment by lukaseder** [comment]
Thanks for your report. I suspect we'll need new `GeneratorStrategy` API to specify how these methods should be named, or alternatively, use the member name instead of the class name for this purpose.

As a workaround, you could post process the generated code with some regular expression search-replace.

**Comment by SightStudio** [comment]
@lukaseder i think creating new GeneratorStrategy API is more reasonable for maintaining backward competability and fluent.
if you're not working on it, can i make a PR fot this?

**Comment by SightStudio** [comment]
I've returned with my solution.

My goal was just add prefix to generated table class.
(like generated QClass of QueryDSL)

So i solved this issue by checking definitionPath contains implementations of AbstractTableDefinition.
(MySQLTableDefinition for mine).

now I got "JPrefixed" table class without having this issue.

```java
public class JPrefixGeneratorStrategy extends DefaultGeneratorStrategy {

    @Override
    public String getJavaClassName(final Definition definition, final Mode mode) {
        if (mode == Mode.DEFAULT) {
            if (isTable(definition)) {
                return "J" + super.getJavaClassName(definition, mode);
            }
        }
        return super.getJavaClassName(definition, mode);
    }

    private boolean isTable(Definition definition) {
        List<Definition> definitionPath = definition.getDefinitionPath();
        for (Definition tableDefinition : definitionPath) {
            if (tableDefinition instanceof AbstractTableDefinition) {
                return true;
            }
        }
        return false;
    }
}
```

**Comment by lukaseder** [comment]
Duplicate of:

- https://github.com/jOOQ/jOOQ/issues/7668

This is implemented in jOOQ 3.20
